handlers/line/line.go

```diff
@@ -1,3 +1,178 @@
 package line
 
-/* no logs */
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/nyaruka/courier/utils"
+	"github.com/nyaruka/gocommon/urns"
+	"net/http"
+	"time"
+
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+)
+
+var sendURL = "https://api.line.me/v2/bot/message/push"
+var maxMsgLength = 1600
+
+func init() {
+	courier.RegisterHandler(newHandler())
+}
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+func newHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("LN"), "Line")}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	return s.AddHandlerRoute(h, http.MethodPost, "receive", h.ReceiveMessage)
+}
+
+// {
+// 	"events": [
+// 	  {
+// 		"replyToken": "nHuyWiB7yP5Zw52FIkcQobQuGDXCTA",
+// 		"type": "message",
+// 		"timestamp": 1462629479859,
+// 		"source": {
+// 		  "type": "user",
+// 		  "userId": "U4af4980629..."
+// 		},
+// 		"message": {
+// 		  "id": "325708",
+// 		  "type": "text",
+// 		  "text": "Hello, world"
+// 		}
+// 	  },
+// 	  {
+// 		"replyToken": "nHuyWiB7yP5Zw52FIkcQobQuGDXCTA",
+// 		"type": "follow",
```

Comment:
Do we not support follows?

Reply:
No

Reply:
Any reason why not?

Reply:
I guess if we don't support it in the past that's ok to leave it off for now.

Reply:
right